packages/core/src/connectivity/scp-cf/destination/destination.ts

```diff
@@ -55,7 +55,54 @@ export function parseDestination(
     }
   );
 
-  return sanitizeDestination(destination);
+  const additionalHeadersAndQueryParameters =
+    getAdditionalHeadersAndQueryParameters(destinationConfig);
+
+  return sanitizeDestination({
+    ...destination,
+    ...additionalHeadersAndQueryParameters
+  });
+}
+
+function getAdditionalProperties(
+  destinationConfig: DestinationConfiguration,
+  originalKeyPrefix: string
+): Record<string, any> {
+  const relevantConfigEntries = Object.entries(destinationConfig).filter(
+    ([key]) => key.startsWith(originalKeyPrefix)
+  );
+
+  return relevantConfigEntries.reduce(
+    (additionalProperties: Record<string, any>, [originalKey, value]) => {
+      const headerKey = originalKey.replace(originalKeyPrefix, '');
+      additionalProperties[headerKey] = value;
+      return additionalProperties;
+    },
+    {}
+  );
+}
+
+function getAdditionalHeadersAndQueryParameters(
+  destinationConfig: DestinationConfiguration
+): Record<string, any> {
```

Comment:
Could you perhaps use a strong type like an interface with the keys headers and query parameters?

Reply:
No, because I  don't know the keys. Those will either be the headers or the query parameters. The keys `headers` and `queryParameter` are not returned here, but wrapped around outside.

Reply:
Done.